sdk/platform/http_client/curl/src/curl_client.cpp

```diff
@@ -32,5 +32,117 @@ Response CurlClient::send()
     }
   }
 
-  return Response(200, "OK\n");
+  return this->m_response;
+}
+
+CURLcode CurlClient::Perform()
+{
+  m_firstHeader = true;
+
+  auto settingUp = SetUrl();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetHeaders();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  settingUp = SetWriteResponse();
+  if (settingUp != CURLE_OK)
+  {
+    return settingUp;
+  }
+
+  return curl_easy_perform(m_pCurl);
+}
+
+static void ParseAndSetFirstHeader(std::string const& header, Response** response)
+{
+  // set response code, http version and reason phrase (i.e. HTTP/1.1 200 OK)
+  auto start = std::find(header.begin(), header.end(), '/');
+  start++; // skip symbol
+  auto end = std::find(start, header.end(), '.');
+  auto mayorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of minor version
+  end = std::find(start, header.end(), ' ');
+  auto minorVersion = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of status code
+  end = std::find(start, header.end(), ' ');
+  auto statusCode = std::stoi(std::string(start, end));
+
+  start = end + 1; // start of reason phrase
+  auto reasonPhrase = std::string(start, header.end() - 2); // remove \r and \n from the end
+
+  // allocate the instance of response to heap
+  *response = new Response(mayorVersion, minorVersion, HttpStatusCode(statusCode), reasonPhrase);
+  (void)response; // avoid warning about not using response
+}
+
+static void ParseHeader(std::string const& header, Response* response)
+{
+  // get name and value from header
+  auto start = header.begin();
+  auto end = std::find(start, header.end(), ':');
+
+  if (end == header.end())
+  {
+    return; // not a valid header or end of headers symbol reached
+  }
+
+  auto headerName = std::string(start, end);
+  start = end + 1; // start value
+
+  auto headerValue = std::string(start, header.end() - 2); // remove \r and \n from the end
+
+  response->AddHeader(headerName, headerValue);
+}
+
+size_t CurlClient::WriteHeadersCallBack(void* contents, size_t size, size_t nmemb, void* userp)
+{
+  size_t const expected_size = size * nmemb;
```

Comment:
This needs to be overflow checked.

Reply:
I think curl would never return a chunk of data with a size overflowing size_t.  That's the reason this callback is called multiple times when the response is too big right?

Reply:
I see, OK, if we know curl has already provided memory of that size in `contents` I agree you can omit the overflow check, but I would comment as to why that is safe.